src/buf.rs

```diff
@@ -73,30 +67,29 @@ impl Buf {
 
             if self.consumed() > 0 { // let's allocate new slice and move
                 let min_size = old_bytes + bytes;
-                assert!(min_size < MAX_BUF_SIZE);
-                let mut vec = Vec::with_capacity(
-                    max(min_size, min(old_cap*2, MAX_BUF_SIZE)));
+
+                let mut vec = Vec::with_capacity(max(min_size, old_cap*2));
```

Comment:
`old_cap*2` should be a `old_cap.saturating_mul(2)` because on 32bit platforms it can easily overflow.